File: Examples/VisualBasic/Pages/PushCurrentPageVersion.vb
Imports System.IO
Imports Aspose.Note
Imports System.Collections.Generic

Namespace Pages
    Public Class PushCurrentPageVersion
        Public Shared Sub Run()
            ' ExStart:PushCurrentPageVersion
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir_Pages()

            ' Load OneNote document and get first child           
            Dim document As New Document(dataDir & Convert.ToString("Aspose.one"))
            Dim page As Page = document.FirstChild

            Dim pageHistory = document.GetPageHistory(page)

            pageHistory.Add(page.Clone())

            document.Save(dataDir & "PushCurrentPageVersion_out.one")
            ' ExEnd:PushCurrentPageVersion
        End Sub
    End Class
End Namespace

File: Examples/CSharp/Pages/PushCurrentPageVersion.cs
using System.IO;
using Aspose.Note;
using System;
using System.Collections.Generic;

namespace Aspose.Note.Examples.CSharp.Pages
{
    public class PushCurrentPageVersion
    {
        public static void Run()
        {
            // ExStart:PushCurrentPageVersion
            // ExFor:Page.Clone
            // ExSummary:Shows how to push current version of a page to history.

            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_Pages();

            // Load OneNote document and get first child           
            Document document = new Document(dataDir + "Aspose.one");
            Page page = document.FirstChild;

            var pageHistory = document.GetPageHistory(page);

            pageHistory.Add(page.Clone());

            document.Save(dataDir + "PushCurrentPageVersion_out.one");

            // ExEnd:PushCurrentPageVersion
        }
    }
}